web/packages/teleport/src/Roles/RoleEditor/RoleEditor.tsx

```diff
@@ -83,6 +88,17 @@ export const RoleEditor = ({
     isDirty: !originalRole, // New role is dirty by default.
   });
 
+  const isDirty = (): boolean => {
```

Comment:
That reminds me that I've recently used an IIFE just to be able to use `switch` within a component like this, though here of course a function works just as well.